it/google-cloud-platform/src/main/java/org/apache/beam/it/gcp/GCPBaseIT.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.it.gcp;
+
+import com.google.api.gax.core.CredentialsProvider;
+import com.google.api.gax.core.FixedCredentialsProvider;
+import com.google.auth.Credentials;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import org.apache.beam.it.common.TestProperties;
+import org.junit.Before;
+
+@SuppressWarnings("nullness")
```

Comment:
Ah, this is interesting. Because it is in `src/main` it is not automatically skipped. [We do have checker disabled in all tests](https://github.com/apache/beam/blob/51512cadf2d65c1e0dc3f19e3287759a1dd78b2f/buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy#L1098), because it does not work well with JUnit APIs.

I don't have enough context on the design of the modules to know if we should change anything here, so I am just sharing this as an FYI that we may want to work on.